2014 Primary Election Alameda test failure

Missing write-in candidates. Not available at precinct-level, but just [totals](https://www.acgov.org/rov/elections/20140603/documents/OfficialWrite-in-Tally.pdf)

```
E           AssertionError: Alameda failed
E           assert {('"Bo" Bogda...): 75182, ...} == {('"Bo" Bogdan...): 75182, ...}
E             Omitting 97 identical items, use -v to show
E             Left contains more items:
E             {('Karen Jill Bernal', 'Alameda', '', 'Governor'): 2}
E             Use -v to get the full diff
```

--- a/2014/20140603__ca__primary__alameda__precinct.csv
+++ b/2014/20140603__ca__primary__alameda__precinct.csv
@@ -56539,3 +56539,4 @@ Alameda,9880200,Treasurer,,REP,Greg Conlon,3
 Alameda,9880200,Treasurer,,DEM,John Chiang,1
 Alameda,9880200,U.S. House,15,DEM,Eric Swalwell,1
 Alameda,9880200,U.S. House,15,REP,Hugh Bussell,3
+Alameda,Write-In,Governor,,DEM,Karen Jill Bernal,2

--- a/src/swdb/candidates/P14.json
+++ b/src/swdb/candidates/P14.json
@@ -152,4 +152,8 @@
  "Jonathan Madison": "Jonathan Emmanuel Madison",
  "Douglas Tuma": "Douglas Arthur Tuma",
  "Janice Bonser": "Janice Marlae Bonser",
- "Phill Tufi": "Phill A. Tufi"}
+ "Phill Tufi": "Phill A. Tufi",
+ "KAREN JILL BERNAL": "Karen Jill Bernal",
+ "JIMELLE L. WALLS": "Jimelle L. Walls",
+ "NICKOLAS WILDSTAR": "Nickolas Wildstar"
+}

--- a/src/swdb/swdb.py
+++ b/src/swdb/swdb.py
@@ -3,6 +3,7 @@ import numpy as np
 import pandas as pd
 
 from util import dbf_to_dataframe, csv_to_dataframe, COUNTIES
+from writeins import WRITEINS
 
 # Output column names for OpenElections format
 fieldnames = ['county', 'precinct', 'office',
@@ -45,6 +46,9 @@ class SWDBResults(object):
         self.county = county
         self.election = election
         self.codes = codes.Codes(codes_fname(election, county), candidate_norm)
+        self.writeins = None
+        if election in WRITEINS and county in WRITEINS[election]:
+            self.writeins = WRITEINS[election][county](candidate_norm)
         self.df = self.__load_dataframe()
 
     def __load_dataframe(self):
@@ -74,6 +78,8 @@ class SWDBResults(object):
         result.votes = result.votes.astype(np.int32)
         result = result.groupby(
             fieldnames[:-1]).sum().reset_index()[fieldnames]
+        if self.writeins:
+            result = result.append(self.writeins)
         for x in ['candidate', 'district', 'office', 'precinct']:
             result = result.sort_values(by=x, kind='mergesort')
         return result

--- a/src/swdb/writeins.py
+++ b/src/swdb/writeins.py
@@ -1,0 +1,23 @@
+import io
+import requests
+import PyPDF2
+
+
+def alameda_p14(candidate_norm):
+    writeins_url = 'https://www.acgov.org/rov/elections/20140603/documents/OfficialWrite-in-Tally.pdf'
+    writeins_req = requests.get(writeins_url)
+    writeins_req.raise_for_status()
+    writeins_pdf = PyPDF2.PdfFileReader(io.BytesIO(writeins_req.content))
+    cands = writeins_pdf.getPage(0).extractText().split('\n')[7::3]
+    cands = [c.strip(' ') for c in cands]
+    cand_split = [(c.split(' ' * 7)[0],
+                   int(c.split(' ' * 7)[1].strip())) for c in cands]
+    return [{'candidate': candidate_norm[c],
+             'votes': v,
+             'county': 'Alameda',
+             'office': 'Governor',
+             'party': 'DEM',
+             'precinct': 'Write-In'} for
+            c, v in cand_split if v != 0]
+
+WRITEINS = {'P14': {'Alameda': alameda_p14}}

